### PR TITLE
Added new GUI and editor themes

### PR DESCRIPTION
**Pull request by nullbasis**
I added Solarized Light and Solarized Dark as both editor and GUI themes since I like to use them myself, and thought I might as well contribute them to the project in case anyone else wants them too.

### CODE REVIEW

File: nw/assets/themes/gui/solarized_dark/theme.conf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+button          =   7,  54,  66
+buttontext      = 253, 246, 227
+brighttext      = 253, 246, 227
+highlight       =  42, 161, 152
+highlightedtext =   0,  43,  54
+link            =  38, 139, 210
+linkvisited     =  38, 139, 210
+
+[GUI]
+treewordcount   = 253, 246, 227
```
Comment (vkbo) [inline nw/assets/themes/gui/solarized_dark/theme.conf:26]: You can just delete the `treewordcount` entries. They are no longer in use.

File: nw/assets/themes/syntax/solarized_dark.conf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+[Syntax]
+background     =   7,  54,  66
+text           = 253, 246, 227
+link           =  38, 139, 210
+headertext     = 147, 161, 161
+headertag      =  42, 161, 152
+emphasis       =  38, 139, 210
+straightquotes = 238, 232, 213
+doublequotes   = 238, 232, 213
+singlequotes   = 238, 232, 213
```
Comment (vkbo) [inline nw/assets/themes/syntax/solarized_dark.conf:18]: It's fine that the `singlequotes` and `doublequotes` colours are the same, but the `straightquotes` is used to indicate that a quoted piece of text has different quote symbols than those specified in the user preferences. It should therefore be a different colour, preferably some tone of red to stand out.

File: nw/assets/themes/syntax/solarized_light.conf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+licenseurl = https://github.com/altercation/solarized/blob/master/LICENSE
+
+[Syntax]
+background     = 253, 246, 227
+text           =   0,  43,  54
+link           =  38, 139, 210
+headertext     =  88, 110, 117
+headertag      =  42, 161, 152
+emphasis       =  38, 139, 210
+straightquotes =   7,  54,  66
```
Comment (vkbo) [inline nw/assets/themes/syntax/solarized_light.conf:16]: Same comment as for the dark theme.

**Review by vkbo — CHANGES_REQUESTED** [review]
Thanks! I just need you to change the colour for straight quotes to something different than the other two as this is an indicator colour to signal that the wrong quote symbols are in use.

**Comment by vkbo** [comment]
Great! Just a minor request for change above.

I've been meaning to add a section in the docs on how to add themes, as the point I made above isn't immediately obvious. I've also been meaning to add the ability to add user-defined themes locally, in for instance `~/.local/share/novelwriter/` (and equivalent on other OSes).

**Comment by nullbasis** [comment]
Oh, sure, I didn't realize what that was for. I despise smart quotes in all their forms, so I always turn them off for myself.

File: nw/assets/themes/syntax/solarized_dark.conf (hunk 2 above)
Comment (nullbasis) [reply nw/assets/themes/syntax/solarized_dark.conf:18]: So, when I turn off Auto-Replace in my preferences, straight quotes still get the red color specified for them in the theme. Should I open an issue for this? When straight quotes are what I'm after I don't think they should stand out like something is wrong.

**Comment by vkbo** [comment]
They are very convenient for other languages where the symbols aren't as easily accessible on the keyboard as the English ones. In my language we use « and » for instance.

Anyway, I noticed now that the quote highlighting colours are also nearly the same as the text. They should stand out as there is a feature to highlight dialogue (which too can be turned off). That feature should also work with the solarized themes. Thecolour should be different from the emphasis colour.

**Comment by nullbasis** [comment]
Yeah, no problem. I have a tendency to go straight for editing color themes instead of searching for options in settings, so... sorry for the hassle!

File: nw/assets/themes/gui/solarized_dark/theme.conf (hunk 1 above)
Comment (nullbasis) [reply nw/assets/themes/gui/solarized_dark/theme.conf:26]: Oops, I missed this note!

**Review by vkbo — APPROVED** [review]
Perfect!

File: nw/assets/themes/syntax/solarized_dark.conf (hunk 2 above)
Comment (vkbo) [reply nw/assets/themes/syntax/solarized_dark.conf]: You can turn off highlighting of quotes in the Editor section of Preferences. There's a separate switch for it. The highlighting is not connected to the auto-replace feature, although the colour is defined by the symbols chosen.

If you do want dialogue highlighting, but don't want the red colour, just set both double quote symbols in the auto-replace tab to straight quotes, and disable the auto-replace switch.

**Pull request merged**